File: course-service/src/main/java/com/lecture/course/entity/MaterialComponent.java
package com.lecture.course.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.*;

import java.math.BigDecimal;

/**
 * 판매 로트의 성분·함량 한 건
 *
 * 성분명을 자유 문자열로 두면 "리튬" / "LITHIUM" / "Li" 가 섞여
 * 추천 서비스의 성분 교집합이 잡히지 않으므로 반드시 enum 으로 강제한다.
 */
@Embeddable
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode
public class MaterialComponent {

    @Enumerated(EnumType.STRING)
    @Column(name = "component_name", nullable = false, length = 30)
    private ComponentName name;

    /** 함량 0~100 */
    @Column(name = "percentage", nullable = false, precision = 5, scale = 2)
    private BigDecimal percentage;

    public enum ComponentName {
        LITHIUM, COBALT, NICKEL, COPPER, ALUMINUM, IRON, MANGANESE,
        PET, PP, PE, PVC, OTHER
    }
}

File: vue-frontend/src/constants/materialComponent.js
// course-service MaterialComponent.ComponentName과 1:1로 맞춘 성분명
export const COMPONENT_LABELS = {
  LITHIUM: '리튬',
  COBALT: '코발트',
  NICKEL: '니켈',
  COPPER: '구리',
  ALUMINUM: '알루미늄',
  IRON: '철',
  MANGANESE: '망간',
  PET: 'PET',
  PP: 'PP',
  PE: 'PE',
  PVC: 'PVC',
  OTHER: '기타',
}

export const COMPONENT_OPTIONS = Object.entries(COMPONENT_LABELS).map(([value, label]) => ({ value, label }))

export function getComponentLabel(raw) {
  if (!raw) return ''
  return COMPONENT_LABELS[raw] || raw
}
